### DISCUSSION TITLE
en/lc/2705/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#2849] [Announcements]
# en/lc/2705/

LeetCode solutions in any programming language

https://leetcode.doocs.org/en/lc/2705/

**Comment by rain84** [comment]
I think there is slight flaw here.
When `obj` is array, a new array is returning,
but when `obj` is object, the input data is changed at the same time

**Reply by yanglbme** [reply]
Hello, the solution has been updated. Thank you for your feedback.

**Reply by rain84** [reply]
Line 8,
I guess, it will be better to filter array before using of `.map(compactObject)`

**Reply by yanglbme** [reply]
You are right. I have made the change. Thank you~

**Reply by rain84** [reply]
Also the same works for lines 11-14
They can be replaced with 
```ts
if (value) {
  acc[key] = compactObject(value);
}
```
